src/gpu/metal/SDL_gpu_metal.m

```diff
@@ -4055,7 +4070,10 @@ static bool METAL_Submit(
 
         // Notify the fence when the command buffer has completed
         [metalCommandBuffer->handle addCompletedHandler:^(id<MTLCommandBuffer> buffer) {
-          SDL_AtomicIncRef(&metalCommandBuffer->fence->complete);
+            SDL_LockMutex(metalCommandBuffer->fence->mutex);
```

Comment:
Again, create a local variable for the fence instead of indirecting it several times in a row.